### ISSUE TITLE
Fields of format %n where n is an integer do not work in the diagram view with "Enable instance API" ON

### ISSUE THREAD

**Issue by mpradovelasco** [OpenModelica/OpenModelica#10846]
System: OMEdit v1.22.0-dev-161-g8eff86a43d (I think 1.22-dev jun 11) on Windows 11.

When "Enable instance API" is ON, in the following capture image, the bloodSrc component should present 5 and 4 as dimensions of the connector arrays (left and right), but it presents 1 and 1 as shown. This bug appears in all components, which work properly when  "Enable instance API" is OFF.

![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/c2a85846-5a89-48cb-b98b-8fcef304bf05)

For sake of clarity, the icon is defined as:
![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/521167d1-03d8-4f9e-b865-db59fca1e640)
and the code of the referred component is:
```
model inertialBloodSource "Ideal inertial blood flow rate source"
  extends Icons.causalVolEntity;
  ...
  parameter Integer nBin = 1 "Number of input perfusion blood volume flow rate vias" 
    annotation(
    Dialog(connectorSizing = true));
  parameter Integer nBout = 1 "Number of output perfusion blood volume flow rate vias" 
    annotation(
   Dialog(connectorSizing = true));
  Interfaces.inputBlood[nBin] inVBlood(redeclare type chemicals = chemicals) "Array of input perfusion blood vias" 
    annotation(
     Placement(visible = true, 
     transformation(origin = {-74, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0), 
     iconTransformation(origin = {-100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Interfaces.outputBlood[nBout] outVBlood(redeclare type chemicals = chemicals) "Array of output perfusion blood vias" 
    annotation(
    Placement(visible = true, 
    transformation(origin = {72, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0), 
    iconTransformation(origin = {100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
    ...
end inertialBloodSource;
```
I must say that in general, the "Enable Instance API" flag improves the behaviour of OMEdit, and several times, when I have seen some potential bug, it uses to be a mistake in my code that was non detected by OMEdit previously. However, I do not find that mistake in this occasion. 

Thanks

**Comment by casella** [comment]
If it looks like a bug, walks like a bug, and crawls like a bug.... it's a bug!

Thanks for reporting!

**Comment by adeas31** [comment]
@mpradovelasco can you please post the complete example? It is difficult to reproduce the issue with incomplete code.

From a quick look, it seems that the `connectorSizing` value is not updated properly.

**Comment by mpradovelasco** [comment]
The full code of the component is
```
model inertialBloodSource "Ideal inertial blood flow rate source"
  extends Icons.causalVolEntity;
  replaceable type chemicals = Properties.chemicalsDummy "Chemicals in Cyborg"
    annotation(choices(
                    choice(redeclare type chemicals = chemicals "standard chemicals selection")));
  parameter SI.VolumeFlowRate[nBout] bloodFR(each displayUnit = "l/min") = fill(1e-5, nBout) 
    "Blood volume flow rate source for each output";
  parameter Integer nBin = 1 "Number of input perfusion blood volume flow rate vias" 
    annotation(
    Dialog(connectorSizing = true));
  parameter Integer nBout = 1 "Number of output perfusion blood volume flow rate vias" 
    annotation(
    Dialog(connectorSizing = true));
  Interfaces.inputBlood[nBin] inVBlood(redeclare type chemicals = chemicals) "Array of input perfusion blood vias" 
    annotation(
     Placement(visible = true, 
     transformation(origin = {-74, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0), 
     iconTransformation(origin = {-100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Interfaces.outputBlood[nBout] outVBlood(redeclare type chemicals = chemicals) "Array of output perfusion blood vias" 
    annotation(
    Placement(visible = true, 
    transformation(origin = {72, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0), 
    iconTransformation(origin = {100, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  SI.Volume V (displayUnit = "l", start = 0.2e-3, fixed=true)  "Blood volume";
  SI.Mass m[chemicals] (each displayUnit = "ug") "Mass of chemical compounds";
  SI.MassConcentration c[chemicals](each displayUnit = "ug/l") "Concentration of chemicals";
initial equation
  assert(nBin > 0, "Number of input perfursion blood vias must not be zero", level = AssertionLevel.error);
  assert(nBout > 0, "Number of output perfusion blood vias must not be zero", level = AssertionLevel.error);
  for ch in chemicals loop
    der(m[ch])=0;
  end for;
equation
   /* Basic Equations related to systemic behaviour - connections */
  for i in 1:nBout, ch in chemicals loop
    outVBlood[i].c[ch] = c[ch]; // Output concentration is equal to internal well-stirred concentration
  end for;
  for i in 1:nBout loop
    outVBlood[i].bvf = bloodFR[i];
  end for;
  /* Equations associated with explicit internal mechanisms */
  for ch in chemicals loop
    c[ch]*V = m[ch];
    der(m[ch]) = sum(inVBlood[i].bvf*inVBlood[i].c[ch] for i in 1:nBin) - 
      sum(outVBlood[i].bvf for i in 1:nBout)*c[ch];
  end for;
  der(V) = sum(inVBlood[i].bvf for i in 1:nBin) - sum(outVBlood[i].bvf for i in 1:nBout);
  annotation(
  defaultComponentName = "ibloodSrc",
  Icon(coordinateSystem(initialScale = 0.1, extent = {{-100, -100}, {100, 100}}), graphics = {Line(origin = {76.4432, 0.272727}, points = {{-11, 0}, {13, 0}}), Line(origin = {-76.2841, 0}, points = {{-15, 0}, {11, 0}}), Line(origin = {-26.81, 0.46}, points = {{14, 0}, {56, 0}}, color = {170, 0, 0}, thickness = 0.5, arrowSize = 8), Ellipse(origin = {-18, 8}, lineColor = {170, 0, 0}, fillColor = {0, 0, 255}, fillPattern = FillPattern.VerticalCylinder, extent = {{-24, 8}, {14, -22}}), Text(origin = {-4, -12}, extent = {{-150, 90}, {150, 50}}, textString = "%name"), Polygon(origin = {-42, 70}, lineColor = {170, 0, 0}, fillColor = {170, 0, 0}, fillPattern = FillPattern.Solid, points = {{90, -70}, {68, -65}, {68, -75}, {90, -70}}), Text(origin = {242, -100}, extent = {{-150, 90}, {-12, 74}}, textString = "%nBout", horizontalAlignment = TextAlignment.Left), Text(origin = {44, -100}, extent = {{-150, 90}, {-12, 74}}, textString = "%nBin", horizontalAlignment = TextAlignment.Left)}));
end inertialBloodSource;
```
The Integer values associated with the connectorSizing annotation are properly assigned by OMEdit to 5 and 4. The but seems appears in the GUI that do not present 5 and 4, but 1 and 1. 

The full code of the model captured in the first image of the issue is:
```
model InertialSource4FLT "PBPK model with 4 flow limited tissues and inertial CO source"
  extends CybSim.ModelsInterfaces.Interfaces.pbpkStr(redeclare type chemicals = chemicals, chemPropDataPBPK = chemicalsProp);
  import CybSim.Properties;
  import PQ = CybSim.BioData.PhysicalChemical;
  /* Local definition of chemicals, mcr, and genes (optional) */
  type chemicals = enumeration(MTX) "Chemical compounds";
  parameter Real chemicalsProp[:, Properties.chemicalProperties] = {{0.001, 454.43928e-3}} "ordered chemical properties per chemical compound";
  /* PBPK and Machines components */
  CybSim.Physiology.PBPK.Ideal.inertialBloodSource bloodSrc(
    redeclare type chemicals = chemicals, bloodFR = {1.66667e-05, 3.33333e-05, 5e-05, 1.66667e-05}, 
    nBin = 5, nBout = 4) annotation(
    Placement(visible = true, transformation(origin = {0, 46}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  CybSim.Physiology.PBPK.Basic.flt flt1(redeclare type chemicals = chemicals, 
    MsBal_RC_m0 = {0}, nBin = 1, nBout = 1) annotation(
    Placement(visible = true, transformation(origin = {0, 24}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  CybSim.Physiology.PBPK.Basic.flt flt2(redeclare type chemicals = chemicals, 
    redeclare class EliminationType = CybSim.Mechanisms.VolRegion.PartialMech.Elimination.clearanceSatPBPK,
    indChoiceElimination = CybSim.Mechanisms.VolRegion.PartialMech.Elimination.EliminationList.clearanceSat, 
    Homogeneous = false, E_cS_Ke = {KeFlt2.so}, PhD_qS_Rt = {0.5}, MsBal_RC_m0 = {0}, nBin = 1, 
    nBout = 1, E_cS_significantPhase = true, redeclare type typVarVolumeFlowRate = input Modelica.Units.SI.VolumeFlowRate "input") annotation(
    Placement(visible = true, transformation(origin = {0, 4}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  CybSim.Physiology.PBPK.Basic.flt flt3(redeclare type chemicals = chemicals, 
    redeclare class ChemicalActivityType = CybSim.Mechanisms.VolRegion.PartialMech.ChemicalActivity.phantomPBPK, 
    indChoiceChemicalActivity = CybSim.Mechanisms.VolRegion.PartialMech.ChemicalActivity.ChemicalActivityList.phantom, 
    ChA_Ph_G = {1.38889e-12}, Vtis0 = 0.005, nBin = 1, nBout = 1) annotation(
    Placement(visible = true, transformation(origin = {0, -16}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  CybSim.Physiology.PBPK.Basic.flt fltDr(redeclare type chemicals = chemicals, 
    Vtis0 = 0.015, nBin = 1, nBout = 1, nDin = 1) annotation(
    Placement(visible = true, transformation(origin = {0, -36}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  CybSim.Machines.DrugSystems.Ideal.idealRate iRate(redeclare type chemicals = chemicals, 
    Tdrug = {900}, drugCh = {chemicals.MTX}, mDose = {{1e-6}}, nDout = 1, widthPulse = {30}) 
    annotation(
    Placement(visible = true, transformation(origin = {-36, -46}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  CybSim.Machines.Pumps.Ideal.extCurrentSource MirrSource(redeclare type chemicals = chemicals "standard chemicals selection", 
    bloodFR = 1e-8, 
    redeclare type typVarMass = input Modelica.Units.SI.MassConcentration "input", c = fltDr.c) 
    annotation(
    Placement(visible = true, transformation(origin = {-62, 30}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
CybSim.Signals.ProcessingSupport.measToSignals KeFlt2(measurement = 5.55556e-06)  annotation(
    Placement(visible = true, transformation(origin = {-76, 78}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
initial equation
/* Computation of physiological Properties */
/* Parametric conversion */
equation
  connect(iRate.outMDrug[1], fltDr.inMDrug[1]) annotation(
    Line(points = {{-26, -46}, {-15, -46}, {-15, -41}, {-6, -41}}, color = {0, 85, 0}));
  connect(bloodSrc.outVBlood[1], flt1.inVBlood[1]) annotation(
    Line(points = {{10, 46}, {26, 46}, {26, 24}, {10, 24}, {10, 24}}, color = {170, 0, 0}));
  connect(bloodSrc.outVBlood[2], flt2.inVBlood[1]) annotation(
    Line(points = {{10, 46}, {26, 46}, {26, 4}, {10, 4}, {10, 4}}, color = {170, 0, 0}));
  connect(bloodSrc.outVBlood[3], flt3.inVBlood[1]) annotation(
    Line(points = {{10, 46}, {26, 46}, {26, -16}, {10, -16}, {10, -16}}, color = {170, 0, 0}));
  connect(bloodSrc.outVBlood[4], fltDr.inVBlood[1]) annotation(
    Line(points = {{10, 46}, {26, 46}, {26, -36}, {10, -36}}, color = {170, 0, 0}));
  connect(flt1.outVBlood[1], bloodSrc.inVBlood[1]) annotation(
    Line(points = {{-10, 24}, {-26, 24}, {-26, 46}, {-10, 46}, {-10, 46}}, color = {170, 0, 0}));
  connect(flt2.outVBlood[1], bloodSrc.inVBlood[2]) annotation(
    Line(points = {{-10, 4}, {-26, 4}, {-26, 46}, {-10, 46}, {-10, 46}}, color = {170, 0, 0}));
  connect(flt3.outVBlood[1], bloodSrc.inVBlood[3]) annotation(
    Line(points = {{-10, -16}, {-26, -16}, {-26, 46}, {-10, 46}, {-10, 46}}, color = {170, 0, 0}));
  connect(fltDr.outVBlood[1], bloodSrc.inVBlood[4]) annotation(
    Line(points = {{-10, -36}, {-26, -36}, {-26, 46}, {-10, 46}}, color = {170, 0, 0}));
/* Additional behaviour code */
  iRate.biInst[1] = false;
  iRate.biPlan[1] = true;
// If iRate is disconnected
//fltDr.inMDrug[1].mf = 0;
//fltDr.inMDrug[1].chem = chemicals.MTX;
  connect(MirrSource.pBlood, bloodSrc.inVBlood[5]) annotation(
    Line(points = {{-52, 30}, {-42, 30}, {-42, 54}, {-10, 54}, {-10, 46}}, color = {170, 0, 0}));
  annotation(
    experiment(StartTime = 0, StopTime = 3600, Tolerance = 1e-06, Interval = 10));
end InertialSource4FLT;
```
The integers associated with the connector dimensions that shoud appear in the diagram view are nBin and nBout. Their values are _(nBin = 5, nBout = 4) for bloodSrc (diagrams shows 1 and 1)_ and _(nBin = 1 , nBout = 1) for flt1, flt2,flt3 and fltDr (diagram shows 1 and 0)_. Therefore, the values shown in diagram are not correct. 
Values of connector dimensions of iRate component are all %nBout, which are properly shown in the model diagram.
Certainly, when Enable Instance API is off, they all are properly shown as the following diagram capture presents:
![imagen](https://github.com/OpenModelica/OpenModelica/assets/114144098/b22d270f-1ac7-4d0d-942c-b008c2f0b3c0)

Thanks!

**Comment by mpradovelasco** [comment]
@adeas31 , thank you very much to build the MWE. I have added the model and tested, but the problem persists for me.

This is the diagram view of the MWE model with Enable Instance API OFF (dimensions are properly presented as 2 or 1):
![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/d958d3ab-72d9-47cd-b8dd-ac9905e1ce8a)
Whereas in the case that the Enable Instance API is ON does not work (all dimensions appear as 1):
![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/b19bc599-bd18-4544-92a8-f68ce907b74f)

I have added new components m or n, and verified this behaviour. I have also unloaded and loaded the MWE to force the update of the diagram. 
I am working on OMEdit v1.22.0-dev-161-g8eff86a43d on Windows 11.
Thanks again.

**Comment by casella** [comment]
@mpradovelasco it looks like the code you are running is not what it claims to be. Do you get the OMC version from Help | About OMEdit?

**Comment by mpradovelasco** [comment]
I am attaching a snapshot of my OMEdit on Windows 11:
![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/1982b815-c0ef-4a04-b068-a97faeaa8512)
I can rebuild OMDev 1.22 with your latest available dev version. It is strange, since the system has a good behaviour in general.

**Comment by mpradovelasco** [comment]
I have uninstalled OMEdit 1.21 (stable versión) to avoid any strange influence (despiste both systems seem work well) and updated de 1.22-dev version to the latest one, but the problem persists as the following capture shows:
![image](https://github.com/OpenModelica/OpenModelica/assets/114144098/d7af8309-3b84-4882-a003-736bbd558c27)

**Comment by adeas31** [comment]
It looks like you are compiling OM instead of using the installer. I see that it is a CMake build.

What is the value of your `%OPENMODELICAHOME%`?

**Comment by casella** [comment]
Yeah, if you really used the nighly build from [here](https://build.openmodelica.org/omc/builds/windows/nightly-builds/64bit/), you should see something like this:
![immagine](https://github.com/OpenModelica/OpenModelica/assets/4233148/42effca2-f879-4929-9a8b-706a1f27005d)
The git hash code should be followed by `(64-bit)`, not by `-cmake`. 

Please try installing the nightly build. If the issue is gone, then we can continue to figure out why you are getting behaviour from older versions when you compile omc yourself, but that is another story 😅

**Comment by adeas31** [comment]
@mpradovelasco PR #10861 fixes this issue. Update to latest version once the PR is merged.

@casella CMake builds are fine and uses the latest version. The issue is more of gcc vs clang. CMake uses gcc by default and Windows makefiles are using clang. There is no sequence point between function arguments and the evaluation order is undefined. I updated the code so don't rely on the evaluation order.

**Comment by mpradovelasco** [comment]
@adeas31, I understand that this issue improvement fixes the issue of using the latest version with CMake in Windows. I will update and build again tomorrow to confirm you the expected behaviour.
@casella, in the case that the build does not work, I will open a separate ticket as you suggest.

Thanks

**Comment by mpradovelasco** [comment]
@adeas31 I have tried to build the OM 1.22 dev today using cmake to check if it works now, and it generates an error when trying to link the OMTLMSimulator. OMDev is updated in my system and the process seem work correctly up to this point. 
Reviewing the instructions, I wonder if I have correct values for the variables used in mingw64 bits. Instructions say (according to the path of my OpenModelica repository: c:\DESARROLLOYDATOS\Areas\Repos\OpenModelica):
```
export OPENMODELICAHOME="c:\\DESARROLLOYDATOS\\Areas\\Repos\\OpenModelica\\build"
export OPENMODELICALIBRARY="c:\\DESARROLLOYDATOS\\Areas\\Repos\\OpenModelica\\build\\lib\\omlibrary"
```
However, cmake generates "build_cmake" folder a not "build". I have tested after change build by buil_cmake, but it does not work. In addition, cmake generates build_cmake/library and not lib. In summary, I wonder if I have problems with these variables definitions for mingw64.
Thanks

**Comment by adeas31** [comment]
@mahge any idea?

**Comment by mahge** [comment]
I am not sure. You can just try to delete the `build_cmake` directory and try again. The issues you suggested do not seem to have anything to do with the changes @adeas31 did.

> However, cmake generates "build_cmake" folder a not "build" 

This is as expected. You can name your build folder anything you want. However, we do not want to name it `build` by default since that is used by the `Autoconf` build system (OpenModelica has two build systems). It is just to allow people to switch between the two build systems without conflicts.

> cmake generates build_cmake/library and not lib

That folder corresponds to the top level `libraries` directory in the OpenModelica source code. This was used for handling Modelica libraries (it is not used anymore but it should not matter for you.)

If you are looking for the `bin`, `lib`, `include`, ... directories that are expected after a completed build and install, then you can find them in `build_cmake/install_cmake`. 

That means, if you have not told CMake to use a specific installation directory, it will use the directory `<build_dir>/install_cmake` directory for installation. You can of course specify another directory if you want to. You can do this by setting `-DCMAKE_INSTALL_PREFIX=<my_install_dir>` when you configure OpenModelica.

**Comment by mpradovelasco** [comment]
I have just built correctly OM 1.22-dev using cmake and instructions of OM Web! It works fine.

The problem was due to define `OMDev` environment variable instead of `OMDEV`. That variable had been removed since the installation of previous nightly build asked for it. 

Thanks
